### PR TITLE
using rangeslider

### PR DESCRIPTION
**Pull request by lc-thomasberger**
fixes:
- ABU-1137
- ABU-758
- ABU-904
- ABU-1136 (Bulletpoint 1)

RTL-support only like previous version

**Important:** test with adapt-contrib-confidenceSlider before merge

### CODE REVIEW

File: js/adapt-contrib-slider.js [hunk 1]
```diff
@@ -422,7 +394,8 @@ define(function(require) {
             if (correctAnswer) {
                 // Check that correctAnswer is neither undefined nor empty
                 answers.push(correctAnswer);
-            } else if (bottom !== undefined) {
+            } else if (bottom && top) {
```
Comment (tomgreenfield) [inline js/adapt-contrib-slider.js]: This if statement probably won't work as expected if `_bottom` has been set to `0`.

**Comment by dancgray** [comment]
The slider-handle is not positioned correctly for model answer after the screen is resized.
screen shot 2016-01-04 at 14 21 20

**Comment by lc-thomasberger** [comment]
@dancgray this was already a bug in the version without the rang-slider library.

**Comment by moloko** [comment]
seeing as the code will get minified by the build process anyway, shall we include the 'unminified' version of the rangeslider code by default? I've occasionally had need to change the source of mediaelement in contrib-media, and having to do so to minified code is a real pain!

**Comment by moloko** [comment]
in your first comment, presumably you meant to say 'LTR-support only like previous version'?
And is that really true? I haven't tested it myself but it claims RTL support in the README

**Comment by chris-steele** [comment]
This is looking good - I am going to look at adding to this the functionality that adapt-contrib-confidenceSlider and adapt-contrib-linkedConfidenceSlider have, as it seems rather unnecessary to have 3 components.

**Comment by moloko** [comment]
the only thing with that is that confidenceSlider and linkedConfidenceSlider are not question components, it therefore always seemed a bit daft to me for them to be coupled so tightly to slider... 
merging linkedConfidenceSlider into confidenceSlider would make a lot of sense though.

**Comment by chris-steele** [comment]
Yep I can agree with that - so I reckon we keep adapt-contrib-confidenceSlider, modify it accordingly to accomodate the improvements to Slider and add the linked capability to it.

**Comment by moloko** [comment]
sounds good to me

**Comment by lc-thomasberger** [comment]
There is a Issue for missing RTL support https://github.com/adaptlearning/adapt_framework/issues/913

**Comment by lc-thomasberger** [comment]
@moloko will add minified code

**Comment by chris-steele** [comment]
I've done the rewrite of adapt-contrib-confidenceSlider, basing it on Thomas' rewrite of adapt-contrib-slider. As discussed I've included the functionality of adapt-contrib-linkedConfidenceSlider. Test package here: https://github.com/chris-steele/sliderMatchingTestPackage. Let me know what you think. Thanks.

**Comment by chris-steele** [comment]
@lc-thomasberger I don't think restoreUserAnswers is working properly. General cause of the issue seems to be trying to manipulate the component DOM before it is created. Could you take a look at this?

**Comment by chris-steele** [comment]
@lc-thomasberger I spoke with @moloko yesterday and given that I've spent a fair bit of time going over the code I am happy to look at fixing the things I mentioned in https://github.com/adaptlearning/adapt-contrib-slider/pull/97#issuecomment-179935236. I'll take a look at this today unless you have any objections?

**Comment by chris-steele** [comment]
OK I've implemented tracking and updated the demo [here](https://github.com/chris-steele/sliderMatchingTestPackage). Also updated the confidence slider [README](https://github.com/chris-steele/sliderMatchingTestPackage/tree/master/src/components/adapt-contrib-confidenceSlider).  There is still some tidy up and optimisation that could be done in the base slider, but I think it's now fully functional and ready for deployment. Let me know what you guys think.

**Comment by hrajotia** [comment]
@dancgray, @lc-thomasberger: I've updated the code for silder-handle having issue on screen re-size.

File: less/slider.less [hunk 2]
```diff
@@ -56,26 +5,36 @@
...
     .slider-scale-numbers {
         position: relative;
-        width: 85%;
-        margin: 45px 5% -7px 5%;
+        width: 100%;
+        margin: 45px 0 -7px 0;
+        .dir-rtl & {
+			transform: rotateY(180deg);
+			-webkit-transform: rotateY(180deg);
+             -ms-transform: rotateY(180deg);
```
Comment (lc-thomasberger) [inline less/slider.less]: i think this is not supported in IE8

**Comment by chris-steele** [comment]
I've replaced rangeslider with the unminified version since minification happens during the build process. Also, I have corrected a line in Plugin.prototype.setPosition so that the entire slider track is filled when the slider is at the maximum position.

**Comment by tomgreenfield** [comment]
Is there any chance we can put any rangeslider modifications as overrides in the slider JS? This makes it as easy as pie to do in-place library upgrades. @oliverfoster is doing very similar in https://github.com/adaptlearning/adapt-contrib-media/pull/139.

**Comment by chris-steele** [comment]
@tomgreenfield sure, stand by for an update...

**Comment by chris-steele** [comment]
@tomgreenfield sorry for the anti-climax, but I don't think it's possible due to the way rangeslider has been written - I can't get at the Plugin prototype. Am I missing a trick?

**Comment by lc-thomasberger** [comment]
once this PR is merged the theme must be updated with this [PR](https://github.com/adaptlearning/adapt-contrib-vanilla/pull/190)

**Comment by tomgreenfield** [comment]
Fair dos @chris-steele.

**Comment by hrajotia** [comment]
+1

**Comment by dancgray** [comment]
Looks great. Just one problem with screen resize when set to RTL. Slider handle re-positions as if it's LTR.

File: README.md [hunk 3]
```diff
@@ -101,7 +103,7 @@ contains values for three types of answers: **correct**, **_incorrect**, and **_
 No known limitations.  
 
 ----------------------------
-**Version number:**  2.0.4   <a href="https://community.adaptlearning.org/" target="_blank"><img src="https://github.com/adaptlearning/documentation/blob/master/04_wiki_assets/plug-ins/images/adapt-logo-mrgn-lft.jpg" alt="adapt learning logo" align="right"></a> 
+**Version number:**  2.0.4   <a href="https://community.adaptlearning.org/" target="_blank"><img src="https://github.com/adaptlearning/documentation/blob/master/04_wiki_assets/plug-ins/images/adapt-logo-mrgn-lft.jpg" alt="adapt learning logo" align="right"></a>
```
Comment (brian-learningpool) [inline README.md]: The version number here should be bumped to 2.1.0 too.

Comment (taylortom) [inline README.md]: Needs bumping

File: less/slider.less [hunk 4]
```diff
@@ -276,4 +129,116 @@
...
+        }
+
+        input[type="range"]:focus + .rangeslider .rangeslider__handle {
+          -moz-box-shadow: none;
+          -webkit-box-shadow: none;
+          box-shadow: none;
+          .accessibility & {
+              outline: 3px solid orange;
+          }
+          /*.dir-rtl & {
```
Comment (taylortom) [inline less/slider.less]: Are we saving this for later?

File: less/slider.less [hunk 5]
```diff
@@ -276,4 +129,116 @@
         text-align: center;
         z-index: 1;
     }
-}
\ No newline at end of file
+}
+
+// reset rangeslider styes
```
Comment (taylortom) [inline less/slider.less]: typo

File: less/slider.less [hunk 6]
```diff
@@ -1,82 +1,39 @@
...
         color: inherit;
-		.dir-rtl & {
-		    /*its supposed to be margin-left and not margin-right due to alignment of slider inside its container*/
-			margin-left: -40px;
-		}
+        margin-left: -25px;
+        .dir-rtl & {
+			transform: rotateY(180deg);
+			-webkit-transform: rotateY(180deg);
+             -ms-transform: rotateY(180deg);
```
Comment (taylortom) [inline less/slider.less]: weird indentation

File: less/slider.less [hunk 7]
```diff
@@ -1,82 +1,39 @@
...
 
     .slider-scale-numbers {
         position: relative;
-        width: 85%;
-        margin: 45px 5% -7px 5%;
+        margin: 45px 21px -7px;
+        .dir-rtl & {
+			transform: rotateY(180deg);
+			-webkit-transform: rotateY(180deg);
+             -ms-transform: rotateY(180deg);
```
Comment (taylortom) [inline less/slider.less]: weird indentation

**Comment by oliverfoster** [comment]
+1

**Comment by tomgreenfield** [comment]
+1

**Comment by brian-learningpool** [comment]
+1

**Pull request merged**